stages/test/test_users.py

```diff
@@ -79,7 +80,7 @@ def test_users_mock_bin(tmp_path, stage_module, user_opts, expected_args):
         }
         options["users"]["foo"].update(user_opts)
 
-        stage_module.main(tmp_path, options)
```

Comment:
Unfortunately after the switch these tests will no longer run as user (and we have no checks that they do, we should probably run all unit tests in CI once as a non-user). The reason is that now "mount" is called here (which was not before). One possible fix is probably to move this test (and the ones below) to mock subprocess.run, something like:
```python
@pytest.mark.parametrize("user_opts,expected_args", TEST_CASES)
@patch("subprocess.run")
def test_users_mock_bin(mocked_run, tmpdir, stage_module, user_opts, expected_args):
    mocked_run.return_value = subprocess.CompletedProcess([], 0)

    make_fake_tree(tmpdir, {
        "/etc/passwd": "",
    })

    options = {
        "users": {
            "foo": {},
        }
    }
    options["users"]["foo"].update(user_opts)

    stage_module.main(tmpdir, options)
    assert mocked_run.call_args_list.count(
        call(["chroot", tmpdir, "useradd"] + expected_args + ["foo"], check=True)) == 1
```

Reply:
Thanks, I'll mock `subprocess.run()`.

Reply:
Fwiw, I opened https://github.com/osbuild/osbuild/pull/1943 that should catch something like this directly in CI (takes about 1-2min to run so seems worth it)